### PR TITLE
NO-ISSUE: Create initial docs about network configuration

### PR DESCRIPTION
**Pull request by mkowalski**
# Assisted Pull Request

## Description

This PR creates a initial documentation explaining different options of
network configuration available when deploying an OCP cluster using
Assisted Service.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @flaper87 
/cc @lranjbar 
/cc @djzager 
/cc @kwozyman 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by djzager** [comment]
Looks like this is a follow-up to #2424

File: docs/user-guide/network-configuration.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+## IP Stack
+
+Currently Assisted Service can deploy OCP clusters using one of the following configurations
+
+* IPv4 with Internet access
+* IPv6 without Internet access (with all the required images mirrored in some local registry)
+* Dual-stack (IPv4 + IPv6)
+
+When using Assisted Service on-premise, configuration of the IP stack is done in the [AgentClusterInstall custom resource](../hive-integration/README.md#AgentClusterInstall). Examples below show can different configurations can be achieved.
```
Comment (kwozyman) [inline docs/user-guide/network-configuration.md:42]: > Examples below show can different configurations can be achieved.

Maybe "show how different configurations"?

**Review by kwozyman — COMMENTED** [review]
As a beginner, I would also like to know what `hostPrefix` parameter is and how to use it

File: docs/user-guide/network-configuration.md (hunk 1 above)
Comment (mkowalski) [reply docs/user-guide/network-configuration.md]: Fixed

File: docs/user-guide/README.md [hunk 2]
```diff
@@ -16,6 +16,10 @@ The assisted-service exposes a RESTFul API which is described in [swagger.yaml](
...
 
-### Using Assisted Service On-Premise
+### Using Assisted Service On-Premises
 
-Please refer to the [Hive Integration readme](../hive-integration/README.md) to learn how to install OCP cluster using Assisted Service on-premise with [Hive](https://github.com/openshift/hive/) and [RHACM](https://github.com/open-cluster-management) (Red Hat Advanced Cluster Management).
+Please refer to the [Hive Integration readme](../hive-integration/README.md) to learn how to install OCP cluster using Assisted Service on-premises with [Hive](https://github.com/openshift/hive/) and [RHACM](https://github.com/open-cluster-management) (Red Hat Advanced Cluster Management).
+
+### Network Configuration
+
+Assisted Service tries to keep feature parity with the OpenShift features regarding advanced network configurations of deployed clusters. Please refer to the [Network Configuration introduction](network-configuration.md) to get more information about how to use those with the Assisted Service.
```
[the remark this answers is not in the document]
Comment (mkowalski) [reply docs/user-guide/README.md]: Done

File: docs/user-guide/network-configuration.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
+# Network Congfiguration
+
+This document is an outline of network configurations supported when deploying OCP using Assisted Service.
```
[the remark this answers is not in the document]
Comment (mkowalski) [reply docs/user-guide/network-configuration.md]: Done

**Review by YuviGold — APPROVED** [review]

**Pull request merged**